src/fsharp/FSharp.Core/list.fs

```diff
@@ -212,15 +212,11 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Fold")>]
         let fold<'T,'State> folder (state:'State) (list: 'T list) = 
-            match list with 
-            | [] -> state
```

Comment:
This optimization for the case where the list is empty has been removed - can it please be added back in?  Or lets just not do this change

Reply:
@dsyme Thanks, I'll add it. The optimization for the empty case is still there to some extent, in the sense that the folding function is not being called in the "empty list" case. But yes, additional check in the beginning for the empty case makes a lot of sense, so the Adapt function is not called prematurely. I'll resubmit.

Reply:
Thanks. I integrated the revert so you can submit a new PR at your leisure :) thanks all!